src/Microsoft.ServiceFabric.Services.Wcf/Remoting/V2/Wcf/Client/WcfServiceRemotingClient.cs

```diff
@@ -3,30 +3,37 @@
 // Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
 // ------------------------------------------------------------
 
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+using System.ServiceModel;
+using System.Threading.Tasks;
+
+using Microsoft.ServiceFabric.Services.Communication.Wcf.Client;
+using Microsoft.ServiceFabric.Services.Remoting.V2.Client;
+using Microsoft.ServiceFabric.Services.Remoting.V2.Messaging;
+
 namespace Microsoft.ServiceFabric.Services.Remoting.V2.Wcf.Client
 {
-    using System;
-    using System.Collections.Generic;
-    using System.Fabric;
-    using System.Globalization;
-    using System.ServiceModel;
-    using System.Threading.Tasks;
-    using Microsoft.ServiceFabric.Services.Communication;
-    using Microsoft.ServiceFabric.Services.Communication.Wcf.Client;
-    using Microsoft.ServiceFabric.Services.Remoting.V2;
-    using Microsoft.ServiceFabric.Services.Remoting.V2.Client;
-    using Microsoft.ServiceFabric.Services.Remoting.V2.Messaging;
-
     internal class WcfServiceRemotingClient : IServiceRemotingClient
     {
         private ServiceRemotingMessageSerializersManager serializersManager;
+        private ExceptionDeserializer exceptionDeserializer;
 
         public WcfServiceRemotingClient(
             WcfCommunicationClient<IServiceRemotingContract> wcfClient,
             ServiceRemotingMessageSerializersManager serializersManager)
         {
             this.serializersManager = serializersManager;
             this.WcfClient = wcfClient;
+
+            IEnumerable<IExceptionConvertor> clientConvertors = new IExceptionConvertor[]

```

Comment:
I think we need to inject the IEnumerable<IExceptionConverter> as a constructor parameter in the WCF Client and Listener classes. I think we should also inject it in the constructor of the respective WCF Factory classes, like we do for FabricTransport Factory classes. This way the WCF API will become consistent with the FabricTransport API from the exception serialization standpoint.

Reply:
By the time we're done, we'll have 4 different places that build a list of well-known converters, fulfilling the rule of 3, so I think this would be a good time to extract into a reusable class or method in the Remoting.dll and verify that the factories using it call into the common logic in unit tests.

Reply:
Injected IEnumerable to factory classes. For now, I haven't extracted the generation of list of well-known converters. We create two lists (one for runtime, one for client) in two different ways so although it is a similar procedure, it is not the same. Let me know if you think that we do want to extract it at this time.